lib/matplotlib/tests/test_labeled_data_unpacking.py

```diff
@@ -0,0 +1,493 @@
+from __future__ import (absolute_import, division, print_function)
+
+from nose.tools import (assert_raises, assert_equal)
+from nose.plugins.skip import SkipTest
+
+try:
+    # 3.2+ versions
+    from nose.tools import assert_regex, assert_not_regex
+except ImportError:
+    try:
+        # 2.7 versions
+        from nose.tools import assert_regexp_matches, assert_not_regexp_matches
+        assert_regex = assert_regexp_matches
+        assert_not_regex = assert_not_regexp_matches
+    except ImportError:
+        # 2.6 versions
+        def noop(txt, regex):
+            raise SkipTest("No assert for regex matching in py2.6")
+        assert_regex = noop
+        assert_not_regex = noop
+
+
+from matplotlib.cbook import is_string_like, iterable
+
+from contextlib import contextmanager
+
+from .. import unpack_labeled_data
+
+
+def is_list_like(obj):
+    return not is_string_like(obj) and iterable(obj)
+
+# Notes on testing the plotting functions itself
+# *   the individual decorated plotting functions are tested in 'test_axes.py'
+# *   that pyplot functions accept a data kwarg is only tested in
+#     test_axes.test_pie_linewidth_0
+
+
+# stolen from pandas
+@contextmanager
+def assert_produces_warning(expected_warning=Warning, filter_level="always",
+                            clear=None):
```

Comment:
Is there a different place where such testing related things should go to? Would also mean that `is_list_like` should go to cbook?